### PR TITLE
refactor: use custom API client wrappers from SDK

### PR DESCRIPTION
**Pull request by craicoverflow**
TL;DR: This PR uses the custom API client contrucutors added to the Golang SDK: https://github.com/redhat-developer/app-services-sdk-go/tree/v0.3.1/kafkamgmt/apiv1#configuration.

This PR also removes the internal client for the Kafka Admin API and now uses the Go SDK client: https://github.com/redhat-developer/app-services-sdk-go/tree/v0.3.1/kafkainstance/apiv1internal

### Verification Steps

Everything should continue to work as normal.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change
- [x] Refactor

### Checklist

- [x] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by craicoverflow** [comment]
@wtrocki @rkpattnaik780 the 77 files changed is mainly deleting `./pkg/api/strimzi-admin` - the main file to focus on is `keycloak_connection.go`.

File: pkg/api/api.go [hunk 1]
```diff
@@ -2,14 +2,14 @@ package api
...
+	kafkainstanceclient "github.com/redhat-developer/app-services-sdk-go/kafkainstance/apiv1internal/client"
 	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
 )
 
 // API is a type which defines a number of API creator functions
 type API struct {
 	Kafka          func() kafkamgmtclient.DefaultApi
 	ServiceAccount func() kafkamgmtclient.SecurityApi
-	TopicAdmin     func(kafkaID string) (strimziadminclient.DefaultApi, *kafkamgmtclient.KafkaRequest, error)
+	KafkaAdmin     func(kafkaID string) (kafkainstanceclient.DefaultApi, *kafkamgmtclient.KafkaRequest, error)
```
Comment (wtrocki) [inline pkg/api/api.go:13]: Thank you for this. FYI - in my branch I'm using registryInstance. We call this KafkaAdmin so it is ok, unless we want to group them somewhat by prefix etc.

**Review by wtrocki — APPROVED** [review]

File: pkg/api/api.go (hunk 1 above)
Comment (craicoverflow) [reply pkg/api/api.go:13]: Since this is just using the KafkaAdmin API, I'd prefer to call it that, but it is easy to change in the future as this would not be used externally.
